com.ibm.wala.core/src/com/ibm/wala/ipa/callgraph/pruned/PrunedCallGraph.java

```diff
@@ -36,9 +36,24 @@
 	private Set<CGNode> keep;
 	private Map<CGNode,Set<CGNode>> remove = Collections.emptyMap();
 	
+	/**
+	 * Create a pruned (filtered) view of an existing call graph.
+	 *
+	 * <p>Note: the created instance retains references to {@code cg} and {@code keep} without making private copies
```

Comment:
Is there a good reason for not just making a private copy?  Would that break existing uses?

Reply:
Making private copies would impose compute and memory demands, of course. So there is a safety/performance trade-off.

Beyond that, making private copies requires cloning the unpruned `cg` call graph deeply enough that the private copy is insulated from outside mutation. None of the `CallGraph` implementations override `Object.clone()`. For most `CallGraph`-implementing classes, I very much doubt that the default `Object.clone()` operates deeply enough to give us a truly private `cg` copy that nobody else can mutate.